main/src/components/PasswordReset.js

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import axios from "axios";
+import { withRouter } from "react-router-dom";
+import hash from "object-hash";
+import "../css/SignupForm.css";
+import { Container, Form, Row, Col, Button } from "react-bootstrap";
+
+class PasswordReset extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      input: {},
+      errors: {},
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+
+    this.setState({
+      input,
+    });
+  }
+
+  async handleSubmit(event) {
```

Comment:
When handling the submit, you will also want to check to see if the new password is the same as the current one.